src/app/account/account/account.component.ts

```diff
@@ -94,4 +102,76 @@ export class AccountComponent implements OnInit {
     this.accountDbService.update(this.accountForm.value);
   }
 
+  facilityDelete() {
+    this.loadingService.setLoadingStatus(true);
+    this.loadingService.setLoadingMessage("Deleting Facility...");
+
+    // Delete all info associated with account
+    this.predictorDbService.deleteAllFacilityPredictors(this.facilityToDelete.id);
+    this.utilityMeterDataDbService.deleteAllFacilityMeterData(this.facilityToDelete.id);
+    this.utilityMeterDbService.deleteAllFacilityMeters(this.facilityToDelete.id);
+    this.utilityMeterGroupDbService.deleteAllFacilityMeterGroups(this.facilityToDelete.id);
+    this.facilityDbService.deleteById(this.facilityToDelete.id);
+
+    // Then navigate to another facility
+    this.facilityDbService.setSelectedFacility();
+    this.loadingService.setLoadingStatus(false);
+    
+  }
+
+  accountDelete() {
+    let selectedAccount: IdbAccount = this.accountDbService.selectedAccount.getValue();
+
+    this.loadingService.setLoadingStatus(true);
+    this.loadingService.setLoadingMessage("Deleting Account...");
+
+    // Delete all info associated with account
+    this.predictorDbService.deleteAllAccountPredictors(selectedAccount.id);
+    this.utilityMeterDataDbService.deleteAllAccountMeterData(selectedAccount.id);
+    this.utilityMeterDbService.deleteAllAccountMeters(selectedAccount.id);
+    this.utilityMeterGroupDbService.deleteAllAccountMeterGroups(selectedAccount.id);
+    this.facilityDbService.deleteAllAccountFacilities();
+    this.accountDbService.deleteById(selectedAccount.id);
+
+    // Then navigate to another account
+    this.accountDbService.setSelectedAccount(1);
```

Comment:
When deleting an account  we wont want to set the selected account with id = 1 because we may delete the account with id = 1 at some point.  You can pass "undefined" to that function and it will set the selected account to the first available account. (if there are no accounts it currently adds one but we will change that eventually to allow all accounts to be deleted).